tokio/src/io/util/read_to_end.rs

```diff
@@ -68,19 +68,33 @@ fn poll_read_to_end<V: VecU8, R: AsyncRead + ?Sized>(
     // of data to return. Simply tacking on an extra DEFAULT_BUF_SIZE space every
     // time is 4,500 times (!) slower than this if the reader has a very small
     // amount of data to return.
-    buf.reserve(32);
+    const NUM_BYTES: usize = 32;
+    let try_small_read = buf.try_small_read_first(NUM_BYTES);
 
     // Get a ReadBuf into the vector.
     let mut read_buf = buf.get_read_buf();
-
     let filled_before = read_buf.filled().len();
-    let poll_result = read.poll_read(cx, &mut read_buf);
-    let filled_after = read_buf.filled().len();
-    let n = filled_after - filled_before;
+    let poll_result;
+
+    let filled_after = if try_small_read {
+        let mut small_buf: [MaybeUninit<u8>; NUM_BYTES] =
+            unsafe { MaybeUninit::uninit().assume_init() };
```

Comment:
You don't need unsafe here.
```suggestion
        let mut small_buf: [MaybeUninit<u8>; NUM_BYTES] = [MaybeUninit::uninit(); NUM_BYTES];
```